[color-deprecations] Deprecated Color Function tests

stacyk opened the pull request:
## Description
Adding tests for https://github.com/sass/sass/blob/colors-2/proposal/color-4-new-spaces.md#deprecated-functions

stacyk committed:
add blackness deprecation warnings

stacyk committed:
add last blackness recommendation

stacyk commented on spec/core_functions/color/blackness.hrx:
```diff
@@ -36,6 +58,17 @@ a {
+<===> middle/zero_whiteness/warning
+DEPRECATION WARNING: Individual color-channel functions defined globally or in the color module are deprecated in favor of the new color.channel() function.
+
+Recommendation: color.channel(color.hwb(0, 0%, 50%), blackness)
```
@mirisuzanne because of https://github.com/sass/sass/blob/colors-2/proposal/color-4-new-spaces.md#modified-color-module-functions I think this would eventually need to be updated to 

`Recommendation: color.channel(hwb(0, 0%, 50%), blackness)`

but when would that need to happen?

stacyk replied:
Also, I don't think the commas are needed (anymore) so I can remove those.

stacyk reviewed (commented):
@mirisuzanne @dvdherron would someone be able to review these examples before @SondraE and I move forward. I don't know if the recommendations are correct, but I like the idea of including them even though I think they may be optional.

stacyk committed:
fix blackness underlines

stacyk committed:
remove color. from hwb, remove commas

dvdherron commented on spec/core_functions/color/blackness.hrx:
```diff
@@ -14,6 +14,17 @@ a {
 }
 
+<===> max/warning
+DEPRECATION WARNING: Individual color-channel functions defined globally or in the color module are deprecated in favor of the new color.channel() function.
```
@stacyk One formatting issue I'm noticing right way is that the text is going over the character limit. I think Sass typically prefers line limit of 80 characters. (The `===` that separate each virtual file is 80 characters wide)

![bild](https://user-images.githubusercontent.com/41588129/201735646-c0361a2b-5af5-4ca7-b41c-8daf6daa624b.png)

stacyk replied:
ah, yes. My editor was wrapping even though I didn't add the line break so I didn't catch this. Thanks @dvdherron

dvdherron commented on spec/core_functions/color/blackness.hrx:
```diff
@@ -14,6 +14,17 @@ a {
+<===> max/warning
+DEPRECATION WARNING: Individual color-channel functions defined globally or in the color module are deprecated in favor of the new color.channel() function.
+
+Recommendation: color.channel(black, blackness)
```
I'm still not sure how this would work for a named color like `black`. 

`channel($color, $channel, $space: null)`

"If channel is not the name of a channel in color, throw an error." The channels here would be red, green, and blue, right? Does some conversion need to happen first?

stacyk replied:
@dvdherron I am not sure. Since the channel is `blackness`, I thought this would be hwb() but maybe it is assumed to be RGB unless the $space is defined as that third param.

dvdherron commented on spec/core_functions/color/blackness.hrx:
```diff
@@ -25,61 +36,127 @@ a {
+
+Recommendation: color.channel(white, blackness)
+
+  ,
```
Seems likely that these warnings will at some point have links where people can get more info about the changes. One thing we will need to find out is whether there is a preference for us to add placeholder lines like "More info: See https://sass-lang.com/d/some-function"

dvdherron reviewed (commented):
@stacyk Based on what I've seen, the format of these look pretty good. For functions being deprecated I think what you and Sondra have here is a good format to continue with. As for what exactly we will use for some of the details . . .we will have to find that out from Natalie or someone else on the Sass team.

I left some minor formatting feedback and a few open questions.

stacyk committed:
add line breaks

SondraE committed:
added warning to all deprecated color functions and removed blackness text that was unnecessary

dvdherron reviewed (approved):
LGTM!

nex3 committed:
[Color 4] Update tests for Color Level 4 (#1846)

This updates most tests for the currently-implemented chunk of the
Color 4 feature. A few tests that need more thorough reworking (such
as being moved out of `error/` for things that are no longer intended
to be errors) are marked as "todo" instead.

See #1828
See sass/sass#2831

dvdherron committed:
Merge branch 'feature.color-4' into color-deprecations

nex3 committed:
Use feature branches for testing the JS API (#1853)

This also skips a few color tests for the embedded host

SondraE committed:
Add tests for color.is-legacy() (#1842)

nex3 committed:
Don't auto-skip dart-sass-embedded along with dart-sass (#1857)

I added this under the erroneous impression that we ran *language*
tests against Dart Sass rather than just *JS API* tests.

SondraE committed:
Add tests for color.space() (#1851)

Co-authored-by: dvdherron <dvdherron@gmail.com>
Co-authored-by: Natalie Weizenbaum <nweiz@google.com>

dvdherron committed:
[Color 4] Update tests for color.grayscale() (#1850)

Co-authored-by: Natalie Weizenbaum <nweiz@google.com>

nex3 committed:
Merge remote-tracking branch 'origin/feature.color-4' into color-deprecations

nex3 committed:
Update specs with Dart Sass

nex3 committed:
Fix a bug where the test runner could deadlock

nex3 committed:
Split up color.hrx
